auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -403,17 +407,111 @@ public static AuthUI getInstance(FirebaseApp app) {
         return R.style.FirebaseUI;
     }
 
+
+    /**
+     * Configuration for an identity provider.
+     *
+     * In the simplest case, you can supply the provider ID and build the config like this:
+     * {@code new IdpConfig.Builder(AuthUI.GOOGLE_PROVIDER).build()}
+     */
+    public static class IdpConfig implements Parcelable {
+        private final String mProviderId;
+        private final List<String> mScopes;
+
+        private IdpConfig(@NonNull String providerId, List<String> scopes) {
+            mScopes = scopes;
+            mProviderId = providerId;
+        }
+
+        public String getProviderId() {
+            return mProviderId;
+        }
+
+        public List<String> getScopes() {
+            return mScopes;
+        }
+
+        protected IdpConfig(Parcel in) {
+            mProviderId = in.readString();
+            mScopes = in.createStringArrayList();
```

Comment:
createStringArrayList actually does read the list, it just handles the creation of the list for you: https://developer.android.com/reference/android/os/Parcel.html#createStringArrayList()